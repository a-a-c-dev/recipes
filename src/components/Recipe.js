import React from "react";
import style from '../recipe.module.css';

const Recipe = React.memo(({ url,title, calories, image, ingredients = [] }) => (
    <div className={style.recipe}>
        <img src={image} className={style.image} alt='' />
        <h3>{title}</h3>
        <h5>Ingredients :</h5>
        <ul>{ingredients.map((x,i) => (<li key={i}>{x.text}</li> ))}
        <a href={url}>See More</a>
        </ul>
        
        <p >Calories: {calories}</p>

    </div>
))

export default Recipe;