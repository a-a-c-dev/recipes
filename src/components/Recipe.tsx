import React from "react";
import style from '../recipe.module.css';

interface RecipeProps { 
    url:string, 
    title:string,
    calories:number, 
    image:string
    ingredients:Ingredient[]
}

interface Ingredient{
    text :string
}


const Recipe = React.memo(({ url,title, calories, image, ingredients = [] }:RecipeProps) => (
    <div role="recipes" className={style.recipe}>
        <img src={image} className={style.image} alt='' />
        <h3 role='title'>{title}</h3>
        <h5>Ingredients :</h5>
        <ul>{ingredients.map((value,i) => (<li role={'ingredients'} key={`${Math.floor(Math.random()*1000000)}-${i}-${value.text}}`}>{value.text}</li> ))}
        <a href={url}>See More</a>
        </ul>
        <p role="calories">Calories: {calories}</p>

    </div>
))

export default Recipe;