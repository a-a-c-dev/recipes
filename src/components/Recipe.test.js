import React from 'react';
import { shallow } from 'enzyme';
import Recipe from './Recipe';
import style from '../recipe.module.css';

describe('MyComponent', () => {
    let wrapper;
    beforeEach(() => {
        wrapper = shallow(<Recipe />);
    });

    afterEach(() => {
        wrapper.unmount();
    });
    it('Should render correctly ', () => {
        expect(wrapper).toMatchSnapshot();
    });
    it('Check the elements on the page', () => {
        expect(wrapper.find('div').length).toBe(1);
        expect(wrapper.find('div').hasClass(style.recipe)).toEqual(true);
        expect(wrapper.find('h5').length).toBe(1);
        expect(wrapper.find('p').length).toBe(1);
        expect(wrapper.find('ol').length).toBe(1);
        expect(wrapper.find('li').length).toBe(0);
        expect(wrapper.find('img').length).toBe(1);
        expect(wrapper.find('img').hasClass(style.image)).toEqual(true);
    })
});
