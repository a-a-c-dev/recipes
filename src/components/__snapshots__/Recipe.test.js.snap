// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MyComponent Should render correctly  1`] = `ShallowWrapper {}`;

exports[`MyComponent Should render correctly in "debug" mode 1`] = `undefined`;

exports[`MyComponent should render correctly in "debug" mode 1`] = `ShallowWrapper {}`;
